store/db_key_layout.go

```diff
@@ -24,32 +24,37 @@ type v1LegacyLayout struct{}
 
 // CalcBlockCommitKey implements BlockKeyLayout.
 func (*v1LegacyLayout) CalcBlockCommitKey(height int64) []byte {
-	return []byte(fmt.Sprintf("C:%v", height))
+	return []byte("C:" + strconv.FormatInt(height, 10))
 }
 
 // CalcBlockHashKey implements BlockKeyLayout.
 func (*v1LegacyLayout) CalcBlockHashKey(hash []byte) []byte {
-	return []byte(fmt.Sprintf("BH:%x", hash))
+	return append([]byte{'B', 'H', ':'}, hash...)
 }
 
 // CalcBlockMetaKey implements BlockKeyLayout.
 func (*v1LegacyLayout) CalcBlockMetaKey(height int64) []byte {
-	return []byte(fmt.Sprintf("H:%v", height))
+	return []byte("H:" + strconv.FormatInt(height, 10))
 }
 
 // CalcBlockPartKey implements BlockKeyLayout.
 func (*v1LegacyLayout) CalcBlockPartKey(height int64, partIndex int) []byte {
-	return []byte(fmt.Sprintf("P:%v:%v", height, partIndex))
+	var (
+		keyPrefix = "P:"
+		keySuffix = strconv.FormatInt(height, 10) + ":" + strconv.Itoa(partIndex)
+		keyStr    = keyPrefix + keySuffix
+	)
+	return []byte(keyStr)
 }
 
```

Comment:
`CalcBlockPartKey` can also be implemented using a `strings.Builder`:
```go
func (*v1LegacyLayout) CalcBlockPartKey(height int64, partIndex int) []byte {
    prefix = "P:"
      
    var sb strings.Builder
    sb.WriteString(prefix)
    sb.WriteString(strconv.FormatInt(x, 10))
    sb.WriteByte(':')
    sb.WriteString(strconv.Itoa(y))
      
    return []byte(sb.String())
}
```
Benchmarks on my laptop show using a `Builder` is ~27% faster then raw string concatenation.
However, the `Builder` allocates 8 bytes per call, whereas raw string concatenation allocates none.

Given that we call this function whenever we load/store block parts (i.e., a lot), I decided to implement raw string concatenation to avoid creating (cumulatively) quite some garbage for the GC to collect.

What do you think? If we prefer speed over memory allocs, I can change this to use a `Builder`.